packages/graphile-build-pg/src/plugins/PgIntrospectionPlugin.js

```diff
@@ -139,393 +205,550 @@ function readFile(filename, encoding) {
   });
 }
 
-export default (async function PgIntrospectionPlugin(
-  builder,
-  {
-    pgConfig,
-    pgSchemas: schemas,
-    pgEnableTags,
-    persistentMemoizeWithKey = (key, fn) => fn(),
-    pgThrowOnMissingSchema = false,
-    pgIncludeExtensionResources = false,
-  }
+/****************************************************************************************
+ * Build Extensions - Add two ways to query the PostgreSQL introspection results to the
+ * build object; one by 'kind' and the other by 'id'. See below for details
+ ****************************************************************************************/
+
+/**
+ * Query the postgres server for the following types of structures in the database:
+ *
+ * namespace - the schemas; from pg_namespace
+ * class - anything `relation-like` (e.g., tables, indices, etc.); from pg_class
+ * attribute - information about table columns; from pg_attribute
+ * type - type information; can be scalars and enums, but are also created for each table in the data base; from pg_type table
+ * constraint - information about the various constraints on the table; from pg_constraints
+ * procedure - information about the aggregate functions as well as custom procedures; from pg_proc
+ * extensions - information about the installed extensions; from pg_extension
```

Comment:
Would be nice to have a link to the relevant section of the PG manual here: https://www.postgresql.org/docs/10/static/catalogs.html